.circleci/config.yml

```diff
@@ -1,89 +1,46 @@
-version: 2.1
-jobs:
-  local_testing:
-    docker:
-      - image: circleci/python:3.9
-    working_directory: ~/project
+version: '2.1'
+
+orbs:
+  python: circleci/python@2.1.1
 
+jobs:
+  build_and_test:
+    executor:
+      name: python/default
+      tag: '3.9'
     steps:
       - checkout
-
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --extras=proxy --verbose
```

Comment:
There are additional optional dependencies in `pyproject.toml` at the moment that need to be installed, like `tenacity`. I'd say either add them here for now or use `--all-extras` here for now to ensure tests are running properly.